tests/contrib/starlette/app.py

```diff
@@ -52,7 +67,56 @@ async def file(request):
         return FileResponse(fp.name)
 
 
-def get_app():
+async def get_tables(request):
+    global engine
+    response = engine.table_names()
+    return PlainTextResponse(str(response))
+
+
+async def list_notes(request):
+    global engine
+    query = "SELECT * FROM NOTES"
+    with engine.connect() as connection:
+        result = connection.execute(query)
+        d, a = {}, []
+        for rowproxy in result:
+            for column, value in rowproxy.items():
+                d = {**d, **{column: value}}
+            a.append(d)
+    response = str(a)
+    return PlainTextResponse(response)
+
+
+async def add_note(request):
+    global engine
```

Comment:
To avoid needing these global definitions everywhere you can define them explicitly at the top of the file.

```python
engine = None
notes_table = None
```

However, I think we should do away with `notes_table` (see my other comment)

Reply:
When I try to define them as None at the top of the file I get a NoneType error :( (even though none of the functions using engine or notes_table are called until after they're set in get_app). However, it works properly with the original global approach.